src/gui/spatialArrow.js

```diff
@@ -0,0 +1,324 @@
+createNameSpace("realityEditor.gui.spatialArrow");
+
+import * as THREE from '../../thirdPartyCode/three/three.module.js';
+
+(function (exports) {
+    
+    let canvasContainer;
+    let canvas;
+    let ctx;
+    let screenW, screenH;
+    let screenRatio;
+    
+    let color, colorLighter;
+
+    function initService() {
+        console.log('%c spatial arrow init!', 'color: green');
+        addCanvas();
+        resizeCanvas();
+        initCanvas();
+        update();
+    }
+
+    // for testing purpose on js fiddle, don't include it in final code
+    const windowRatio = 1;
+
+    window.addEventListener('resize', () => {
+        resizeCanvas();
+        initCanvas();
+        update();
+    });
+
+    // todo: dynamically set style.top based on menu bar's height -- making the arrow canvas right below the menu bar
+    const menuBarHeight = 30;
+
+    function addCanvas() {
+        canvasContainer = document.createElement('div');
+        canvasContainer.className = 'arrow-canvas-container';
+        canvasContainer.style.position = 'absolute';
+        canvasContainer.style.top = '0';
+        canvasContainer.style.left = '0';
+        document.body.appendChild(canvasContainer);
+
+        canvas = document.createElement('canvas');
+        canvas.className = 'arrow-canvas';
+        canvas.style.position = 'absolute';
+        canvas.style.top = `${menuBarHeight}px`;
+        canvas.style.left = '0';
+        canvas.style.zIndex = '3001';
+        canvas.style.border = '2px solid black';
```

Comment:
Is there a reason to include the black border around the edge of the screen, or was that just for debugging?

Reply:
Yea this is just for debugging. Just removed it.